Source/Scene/Model.js

```diff
@@ -3319,37 +3311,46 @@ define([
         };
     }
 
-    function createClippingPlanesEnabledFunction(model) {
-        return function() {
-            return model.clippingPlanesEnabled;
-        };
-    }
-
     function createClippingPlanesLengthFunction(model) {
         return function() {
             return model._packedClippingPlanes.length;
         };
     }
 
-    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0.0);
     function createClippingPlanesFunction(model) {
         return function() {
-            var planes = model.clippingPlanes;
+            var clippingPlanes = model.clippingPlanes;
             var packedPlanes = model._packedClippingPlanes;
-            var length = packedPlanes.length;
-            for (var i = 0; i < length; ++i) {
-                var plane = planes[i];
-                var packedPlane = packedPlanes[i];
 
-                Plane.transform(plane, model._modelViewMatrix, scratchPlane);
-
-                Cartesian3.clone(scratchPlane.normal, packedPlane);
-                packedPlane.w = scratchPlane.distance;
+            if (defined(clippingPlanes) && clippingPlanes.enabled) {
+                clippingPlanes.transformAndPackPlanes(model._modelViewMatrix, packedPlanes);
             }
+
             return packedPlanes;
         };
     }
 
+    function createClippingPlanesEdgeWidthFunction(model) {
+        return function() {
+            if (!defined(model.clippingPlanes)) {
+                return 0.0;
+            }
+
+            return model.clippingPlanes.edgeWidth;
+        };
+    }
+
+    var scratchCartesian = new Cartesian4();
+    function createClippingPlanesEdgeColorFunction(model) {
+        return function() {
+            if (!defined(model.clippingPlanes)) {
+                return scratchCartesian;
+            }
+
+            return Cartesian4.fromColor(model.clippingPlanes.edgeColor, scratchCartesian);
```

Comment:
I'm pretty sure a uniform can be a color instead of being converted to a cartesian. Same comment for globe and point cloud code.